src/pages/githubAssessments/GitHubMissions.tsx

```diff
@@ -0,0 +1,188 @@
+    missionRepo.repoOwner,
+    missionRepo.repoName,
+    '/METADATA',
+    octokit
+  );
+  const browsableMission = createBrowsableMission(missionRepo, metadata);
+
+  return browsableMission;
+}
+
+class BrowsableMission {
+  title: string = '';
+  coverImage: string = '';
+  webSummary: string = '';
+  missionRepoData: MissionRepoData = new MissionRepoData('', '', '');
+  dueDate: Date = new Date(8640000000000000);
+}
+
+function createBrowsableMission(missionRepo: MissionRepoData, metadata: string) {
+  const browsableMission = new BrowsableMission();
+
+  browsableMission.missionRepoData = missionRepo;
+
+  const stringProps = ['coverImage', 'title', 'webSummary'];
+  const dateProps = ['dueDate'];
+
+  const retVal = parseMetadataProperties<BrowsableMission>(
+    browsableMission,
+    stringProps,
+    [],
+    dateProps,
+    metadata
+  );
+
+  return retVal;
+}
+
+function convertMissionToCard(
+  missionRepo: BrowsableMission,
+  octokit: Octokit,
+  isMobileBreakpoint: boolean
+) {
+  const ratio = isMobileBreakpoint ? 5 : 3;
+  const ownerSlashName =
+    missionRepo.missionRepoData.repoOwner + '/' + missionRepo.missionRepoData.repoName;
+  const dueDate = missionRepo.dueDate.toDateString();
+
+  const hasDueDate = new Date(8640000000000000) > missionRepo.dueDate;
+  const isOverdue = new Date() > missionRepo.dueDate;
+  const buttonText = isOverdue ? 'Review Answers' : 'Open';
+
+  const data = missionRepo.missionRepoData;
+
+  return (
+    <div key={ownerSlashName}>
+      <Card className="row listing" elevation={Elevation.ONE}>
+        <div className={`col-xs-${String(ratio)} listing-picture`}>
+          <img
+            alt="Assessment"
+            className={`cover-image-${missionRepo.title}`}
+            src={missionRepo.coverImage ? missionRepo.coverImage : defaultCoverImage}
+          />
+        </div>
+
+        <div className={`col-xs-${String(12 - ratio)} listing-text`}>
+          <div className="listing-header">
+            <Text ellipsize={true}>
+              <H4 className="listing-title">{missionRepo.title}</H4>
+              <H6>{ownerSlashName}</H6>
+            </Text>
+          </div>
+
+          <div className="listing-description">
+            <Markdown content={missionRepo.webSummary} />
+          </div>
+
+          <div className="listing-footer">
+            <Text className="listing-due-date">
+              <Icon className="listing-due-icon" iconSize={12} icon={IconNames.TIME} />
+              {hasDueDate ? 'Due: ' + dueDate : 'No due date'}
+            </Text>
+            <div className="listing-button">
+              <Button icon={IconNames.PLAY} minimal={true} onClick={loadIntoEditor}>
+                <span className="custom-hidden-xxxs">{buttonText}</span>
+              </Button>
+            </div>
+          </div>
+        </div>
+      </Card>
+    </div>
+  );
+
+  function loadIntoEditor() {
+    history.push(`/githubassessments/editor`, data);
+  }
```

Comment:
@angelsl im not sure if there's a better way of handling this. it seems a little wrong

Reply:
@chownces Thanks for linking me here. This is actually okay, the rest of the frontend uses this too, my only concern is that I think this will break if the user does a manual refresh (Ctrl+R/F5)..

The other uses in the frontend don't use the history.state, but rather just have the assessment ID in the URL.